tests/ConfigurationSchemaGenerator.Tests/GeneratorTests.cs

```diff
@@ -51,56 +118,1110 @@ public void ShouldRemoveIndentation(string value, string expected)
+            }
+
+            public record ChildType
+            {
+                public string? ChildValue { get; set; }
+
+                // Simple read-only property, ignored.
+                public bool HasParent => Parent != null;
+
+                // Circular reference, included but without type.
+                public ChildType? Parent { get; set; }
+
+                // Ignored via exclusionPaths.
+                public X509Certificate2? ClientCertificate { get; set; }
+            }
+            """;
+
+        ImmutableArray<MetadataReference> references = [
+            MetadataReference.CreateFromFile(typeof(Uri).Assembly.Location),
+            MetadataReference.CreateFromFile(typeof(X509Certificate2).Assembly.Location)
+        ];
+        var schema = GenerateSchemaFromCode(source, references);
+
+        AssertIsJson(schema,
+          $$"""
+          {
+            "type": "object",
+            "properties": {
+              "TestComponent": {
+                "type": "object",
+                "properties": {
+                  "TestProperty": {
+                    {{expected}}
+                  }
+                }
+              }
+            }
+          }
+          """);
+    }
+
+    [Fact]
+    public void ShouldAllowGetOnlyObject()
+    {
+        var source =
+            """
+            [assembly: Aspire.ConfigurationSchema("TestComponent", typeof(TestSettings))]
+
+            public class TestSettings
+            {
+                public JsonFormattingOptions FormatSettings { get; } = new();
+            }
+
+            public class JsonFormattingOptions
+            {
+                public bool WriteIndented { get; set; }
+            }
+            """;
+
+        var schema = GenerateSchemaFromCode(source, []);
+
+        Assert.Contains("FormatSettings", schema);
+    }
+
+    [Fact]
+    public void ShouldAllowGetOnlyCollections()
+    {
+        var source =
+            """
+            [assembly: Aspire.ConfigurationSchema("TestComponent", typeof(TestSettings))]
+
+            public class TestSettings
+            {
+                public ICollection<int> CollectionOfInt { get; } = [];
+                public IDictionary<string, int> DictionaryOfStringToInt { get; } = new Dictionary<string, int>();
+            }
+            """;
+
+        var schema = GenerateSchemaFromCode(source, []);
+
+        Assert.Contains("CollectionOfInt", schema);
+        Assert.Contains("DictionaryOfStringToInt", schema);
+    }
+
+    [Fact]
+    public void ShouldAllowFreeFormatViaRecursiveSubtree()
+    {
+        // Example appsettings.json:
+        // {
+        //   "TestComponent": {
+        //     "TestProperty": {
+        //       "UserName": "john.doe@email.com",
+        //       "Password": "P@ssw0rd!",
+        //       "Options": {
+        //         "RequireSSL": "false"
+        //       }
+        //     }
+        //   }
+        // }
```

Comment:
My biggest concern is that this comment doesn't match the code below. I don't see it using `TestProperty1` or `TestProperty2`. I think the comment is confusing since it doesn't match.

Reply:
Makes sense. I've replaced the comment with compilable code.